packages/test/test-end-to-end-tests/src/test/gc/gcTombstone.spec.ts

```diff
@@ -0,0 +1,294 @@
+        TestDataObject,
+        [],
+        [],
+    );
+    const inactiveTimeoutMs = 0;
+    const wait = 100;
+    const sweepTimeoutMs = 200;
+
+    const gcOptions: IGCRuntimeOptions = {
+        gcAllowed: true,
+        sweepAllowed: true,
+        snapshotCacheExpiryMs: 0,
+        inactiveTimeoutMs,
+    };
+
+    const runtimeOptions: IContainerRuntimeOptions = {
+        summaryOptions: {
+            summaryConfigOverrides: {
+                state: "disableHeuristics",
+                maxAckWaitTime: 10000,
+                maxOpsSinceLastSummary: 7000,
+                initialSummarizerDelayMs: 0,
+                summarizerClientElection: false,
+            },
+        },
+        gcOptions,
+    };
+
+    const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+        runtime.IFluidHandleContext.resolveHandle(request);
+
+    const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+        dataObjectFactory,
+        [
+            [dataObjectFactory.type, Promise.resolve(dataObjectFactory)],
+        ],
+        undefined,
+        [innerRequestHandler],
+        runtimeOptions,
+    );
+
+    const settings = {
+        "Fluid.GarbageCollection.Test.Tombstone": "true",
+        "Fluid.GarbageCollection.TestOverride.SweepTimeoutMs": sweepTimeoutMs,
+    };
+    const configProvider = mockConfigProvider(settings);
+    const createContainer = async () => provider.createContainer(runtimeFactory, { configProvider });
+    const loadSummarizerAndContainer = async (summaryVersion?: string) => {
+        const absoluteUrl = await mainContainer.getAbsoluteUrl("");
+        return createSummarizerWithContainer(provider, absoluteUrl, runtimeFactory, { configProvider }, summaryVersion);
+    };
+    const summarize = async (summarizer: ISummarizer) => {
+        await provider.ensureSynchronized();
+        return summarizeNow(summarizer);
+    };
+
+    let mainContainer: IContainer;
+    let mainDataStore: TestDataObject;
+
+    beforeEach(async function() {
+        provider = getTestObjectProvider({ syncSummarizer: true });
+        if (provider.driver.type !== "local") {
+            this.skip();
+        }
+        mainContainer = await createContainer();
+        mainDataStore = await requestFluidObject<TestDataObject>(mainContainer, "default");
+        await waitForContainerConnection(mainContainer);
+    });
+
+    const createUnreferencedDataStore = async () => {
+        const handleKey = "handle";
+        const testDataObject = await dataObjectFactory.createInstance(mainDataStore.containerRuntime);
+        const testDataObjectId = testDataObject.id;
+
+        // Reference a datastore and summarize
+        mainDataStore._root.set(handleKey, testDataObject.handle);
+        const {
+            container: summarizingContainer1,
+            summarizer: summarizer1,
+        } = await loadSummarizerAndContainer();
+        await summarize(summarizer1);
```

Comment:
You don't need this step. You can add the handle to a DDS, then remove the handle and summarize. It will achieve the same effect and make the test run faster.